packages/aws-cdk-lib/aws-efs/lib/efs-file-system.ts

```diff
@@ -579,6 +651,19 @@ export class FileSystem extends FileSystemBase {
       replicationOverwriteProtection: props.replicationOverwriteProtection,
     } : undefined;
 
+    const replicationConfiguration = enable ? {
+      destinations: [
+        {
+          fileSystemId: destinationFileSystem?.fileSystemId,
+          kmsKeyId: kmsKey?.keyArn,
+          region: region ??
+            // if destinationFileSystem is set, specify the region of the destination file system
+            (destinationFileSystem ? destinationFileSystem.env.region : Stack.of(this).region),
```

Comment:
```suggestion
          region: destinationFileSystem ? destinationFileSystem.env.region : (region ?? Stack.of(this).region)
```
Should work the same, but the logic it's closer to the above conditions.